server/models/userModel.js

```diff
@@ -0,0 +1,60 @@
+const { DataTypes: dt } = require("sequelize");
+const crypto = require("crypto");
+const db = require("./index");
+
+const UserModel = db.define("User", {
+	id: {
+		type: dt.INTEGER,
+		primaryKey: true,
+		autoIncrement: true,
+	},
+	name: {
+		type: dt.STRING,
+		allowNull: false,
+	},
+	email: {
+		type: dt.STRING,
+		allowNull: false,
+		unique: true,
+	},
+	password: {
+		type: dt.STRING,
+		allowNull: false,
+		//Prevents this field from showing up in findAll and findById queries
+		get() {
+			return () => this.getDataValue("password");
+		},
+	},
+	salt: {
+		type: dt.STRING,
+		//Prevents this field from showing up in findAll and findById queries
+		get() {
+			return () => this.getDataValue("salt");
+		},
+	},
+});
+
+const hashAndSalt = function (password, salt) {
+	return crypto
+		.createHash("RSA-SHA256")
+		.update(password)
+		.update(salt)
+		.digest("hex");
+};
+
+const setSaltAndPassword = function (user) {
+	if (user.changed("password")) {
+		user.salt = crypto.randomBytes(16).toString("base64");
+		user.password = hashAndSalt(user.password(), user.salt());
+	}
+};
+
+UserModel.beforeCreate(setSaltAndPassword);
+UserModel.beforeUpdate(setSaltAndPassword);
```

Comment:
Is this duplicate purposeful (to simulate "rounds")? 

If it isn't, that's something to correct before merging. 

If it is, you might consider using a package like `bcrypt` which allow you to easily do salting rounds. Doing just two salting rounds is not really much more secure than doing just one.